File: Python/Program/Vehicle.py
'''
Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 4 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
'''

class Vehicle:
    # Private Attributes
    __platNomor = ""
    __merk = ""
    __tahunProduksi = ""
    __warna = ""
    __kapasitasTangki = ""
    
    
    # Constructor
    def __init__(self, platNomor="", merk="", tahunProduksi="", warna="", kapasitasTangki=""):
        self.__platNomor = platNomor
        self.__merk = merk
        self.__tahunProduksi = tahunProduksi
        self.__warna = warna
        self.__kapasitasTangki = kapasitasTangki
        
    # Getter and Setter
    # Getter 
    def getPlatNomor(self):
        return self.__platNomor
    
    def getMerk(self):
        return self.__merk
    
    def getTahunProduksi(self):
        return self.__tahunProduksi
    
    def getWarna(self):
        return self.__warna
    
    def getKapasitasTangki(self):
        return self.__kapasitasTangki
    
    # Setter
    def setPlatNomor(self, platNomor):
        self.__platNomor = platNomor
    
    def setMerk(self, merk):
        self.__merk = merk
    
    def setTahunProduksi(self, tahunProduksi):
        self.__tahunProduksi = tahunProduksi
    
    def setWarna(self, warna):
        self.__warna = warna
        
    def setKapasitasTangki(self, kapasitasTangki):
        self.__kapasitasTangki = kapasitasTangki
    

File: Cpp/Program/Vehicle.cpp
/*Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 4 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/
#pragma once
#include <iostream>
#include <string>

using namespace std;

class Vehicle{

    /*Attribute*/
    private:
        string platNomor;
        string merk;
        string tahunProduksi;
        string warna;
        string kapasitasTangki;


    /*Constructor*/
    public:
        Vehicle()
        {}

        Vehicle(string platNomor, string merk, string tahunProduksi, string warna, string kapasitasTangki){
            this->platNomor = platNomor;
            this->merk = merk;
            this->tahunProduksi = tahunProduksi;
            this->warna = warna;
            this->kapasitasTangki = kapasitasTangki;
        }

        /*Getter & Setter*/
        //Getters
        string getPlatNomor() {
            return platNomor;
        }
        
        string getMerk() {
            return merk;
        }
        
        string getTahunProduksi() {
            return tahunProduksi;
        }
        
        string getWarna() {
            return warna;
        }
        
        string getKapasitasTangki() {
            return kapasitasTangki;
        }
        
        // Setters
        void setPlatNomor(string platNomor) {
            this->platNomor = platNomor;
        }
        
        void setMerk(string merk) {
            this->merk = merk;
        }
        
        void setTahunProduksi(int tahunProduksi) {
            this->tahunProduksi = tahunProduksi;
        }
        
        void setWarna(string warna) {
            this->warna = warna;
        }
        
        void setKapasitasTangki(float kapasitasTangki) {
            this->kapasitasTangki = kapasitasTangki;
        }

        virtual ~Vehicle(){}
};